core/src/main/scala/com/acervera/osm4scala/model/NodeEntity.scala

```diff
@@ -28,6 +28,35 @@ package com.acervera.osm4scala.model
 /**
   * Entity that represent a OSM node as https://wiki.openstreetmap.org/wiki/Elements#Node and https://wiki.openstreetmap.org/wiki/Node describe
   */
-case class NodeEntity(id: Long, latitude: Double, longitude: Double, tags: Map[String, String]) extends OSMEntity {
+case class NodeEntity(id: Long,
+                      latitude: Double,
+                      longitude: Double,
+                      tags: Map[String, String],
+                      version: Option[Int],
+                      timestamp: Option[Long],
+                      changeset: Option[Long],
+                      uid: Option[Int],
+                      user: Option[String],
+                      visible: Option[Boolean]) extends OSMEntity {
   override val osmModel: OSMTypes.Value = OSMTypes.Node
+
+  def apply(id: Long,
+            latitude: Double,
+            longitude: Double,
+            tags: Map[String, String]): NodeEntity = {
+    NodeEntity(id, latitude, longitude, tags,
+      None, None, None, None, None, None)
+  }
+
+  override def toString: String = {
```

Comment:
The case classes have a good enough toString implementation ([maybe not the best](https://github.com/scala/scala/pull/6936)). Please, don't overwrite it if it is not necessary for the bussines logic.

Reply:
It wouldn't deal with array and nested structure well. Printing everything explicitly out helps debugging. If you truly think no need, I can remove these.